snuba/migrations/snuba_migrations/metrics/templates.py

```diff
@@ -190,21 +214,35 @@ def get_forward_view_migration_local(
     mv_name: str,
     aggregation_col_schema: Sequence[Column[Modifiers]],
     aggregation_states: str,
-    granularity: int,
+    granularity: Optional[int],
 ) -> operations.SqlOperation:
     aggregated_cols = [*COMMON_AGGR_COLUMNS, *aggregation_col_schema]
-    return operations.CreateMaterializedView(
-        storage_set=StorageSetKey.METRICS,
-        view_name=mv_name,
-        destination_table_name=table_name,
-        columns=aggregated_cols,
-        query=MATVIEW_STATEMENT
-        % {
-            "raw_table_name": source_table_name,
-            "aggregation_states": aggregation_states,
-            "granularity": granularity,
-        },
-    )
+
+    if granularity:
+        return operations.CreateMaterializedView(
+            storage_set=StorageSetKey.METRICS,
+            view_name=mv_name,
+            destination_table_name=table_name,
+            columns=aggregated_cols,
+            query=MATVIEW_STATEMENT
+            % {
+                "raw_table_name": source_table_name,
+                "aggregation_states": aggregation_states,
+                "granularity": granularity,
+            },
+        )
+    else:
+        return operations.CreateMaterializedView(
+            storage_set=StorageSetKey.METRICS,
+            view_name=mv_name,
+            destination_table_name=table_name,
+            columns=aggregated_cols,
+            query=MATVIEW_STATEMENT_CONSOLIDATED
+            % {
+                "raw_table_name": source_table_name,
+                "aggregation_states": aggregation_states,
+            },
+        )
```

Comment:
Why not just copying this piece of code in your forward_local methods.
Migrations were meant to be isolated from each other and immutable as they are append only. 
The idea of having complex templates that generate them has a lot of drawbacks as it is easy to change the semantic of an existing migration by mistake by refactoring the template. And it will be very hard to notice.